Sources/Datadog/Datadog+Internal.swift

```diff
@@ -6,26 +6,38 @@
 
 import Foundation
 
-/// This class exposes internal methods that are used by other Datadog modules and cross platform
+extension Datadog: DatadogInternal {}
+extension Datadog.Configuration.Builder: DatadogInternal {}
+
+/// This extension exposes internal methods that are used by other Datadog modules and cross platform
 /// frameworks. It is not meant for public use.
 ///
-/// DO NOT USE this class or its methods if you are not working on the internals of the Datadog SDK
+/// DO NOT USE this extension or its methods if you are not working on the internals of the Datadog SDK
 /// or one of the cross platform frameworks.
 ///
 /// Methods, members, and functionality of this class  are subject to change without notice, as they
 /// are not considered part of the public interface of the Datadog SDK.
-public class _InternalProxy {
-    public let _configuration = _ConfigurationProxy()
-    public let _telemetry = _TelemetryProxy()
-}
+extension DatadogExtension where ExtendedType: Datadog {
+    /// Internal telemtry proxy.
+    public static var telemetry: _TelemetryProxy { .init() }
 
-public class _TelemetryProxy {
     public func setTelemetryConfigurationMapper(mapper: @escaping (TelemetryConfigurationEvent) -> TelemetryConfigurationEvent) {
         if let rumTelemetry = DD.telemetry as? RUMTelemetry {
             rumTelemetry.configurationEventMapper = mapper
         }
     }
 
+    /// Changes the `version` used for [Unified Service Tagging](https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging).
+    public static func set(customVersion: String) {
+        guard let core = defaultDatadogCore as? DatadogCore else {
+            return
```

Comment:
Shouldn't we add a warning or telemetry event in this case?

Reply:
This code was just moved. Currently, if `defaultDatadogCore` is not `DatadogCore`, it means that the SDK is not initialised. This interface is made for RN to update the version at runtime, a warning won't be actionable by the user, and a telemetry won't get through since the sdk is not initialized.